flink-filesystems/flink-s3-fs-hadoop/src/main/java/org/apache/flink/fs/s3hadoop/S3FileSystemFactory.java

```diff
@@ -94,6 +94,11 @@ protected URI getInitURI(URI fsUri, org.apache.hadoop.conf.Configuration hadoopC
     @Override
     protected S3AccessHelper getS3AccessHelper(FileSystem fs) {
         final S3AFileSystem s3Afs = (S3AFileSystem) fs;
-        return new HadoopS3AccessHelper(s3Afs, s3Afs.getConf());
+        return new HadoopS3AccessHelper(
+                s3Afs,
+                s3Afs.getConf(),
+                s3Afs.createStoreContext().getInstrumentation(),
```

Comment:
we can probably do the extraction inside the HadoopS3AccessHelper constructor without changing here and the constructor signature.

Reply:
Makes sense. Changed it.